tests/components/modbus/test_modbus_binary_sensor.py

```diff
@@ -0,0 +1,98 @@
+"""The tests for the Modbus sensor component."""
+import logging
+
+from homeassistant.components.binary_sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.components.modbus.const import (
+    CALL_TYPE_COIL,
+    CALL_TYPE_DISCRETE,
+    CONF_ADDRESS,
+    CONF_INPUT_TYPE,
+    CONF_INPUTS,
+)
+from homeassistant.const import CONF_NAME, STATE_OFF, STATE_ON
+
+from .conftest import run_base_test
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def run_sensor_test(hass, use_mock_hub, register_config, value, expected):
+    """Run test for given config."""
+    sensor_name = "modbus_test_binary_sensor"
+    entity_domain = SENSOR_DOMAIN
+    data_array = {
+        CONF_INPUTS: [
+            dict(**{CONF_NAME: sensor_name, CONF_ADDRESS: 1234}, **register_config)
+        ]
+    }
+    await run_base_test(
+        sensor_name,
+        hass,
+        use_mock_hub,
+        data_array,
+        register_config.get(CONF_INPUT_TYPE),
+        entity_domain,
+        value,
+        expected,
+    )
+
+    # Check state
+    entity_id = f"{entity_domain}.{sensor_name}"
+    state = hass.states.get(entity_id).state
+    assert state == expected
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6243615)